src/platform/Linux/DnssdImpl.cpp

```diff
@@ -743,20 +748,39 @@ void MdnsAvahi::HandleBrowse(AvahiServiceBrowser * browser, AvahiIfIndex interfa
     case AVAHI_BROWSER_ALL_FOR_NOW: {
         ChipLogProgress(DeviceLayer, "Avahi browse: all for now");
         bool needRetries = context->mBrowseRetries++ < kMaxBrowseRetries && !context->mStopped.load();
+
         // If we were already asked to stop, no need to send a callback - no one is listening.
         if (!context->mStopped.load())
         {
             context->mCallback(context->mContext, context->mServices.data(), context->mServices.size(), !needRetries,
                                CHIP_NO_ERROR);
+
+            // Clearing records/services already passed to application through delegate. Keeping it may cause
```

Comment:
But the application will anyway receive duplicates if we do a retry, right?
The question is, should we even do any retries if we do a continuous query? Shouldn't we let avahi just keep listening without sending more active queries?

Reply:
Here in same delegates we were sending duplicate records. delegate get called first time with record x without duplicates, the next delegate call with one duplicate and the following with 2 and so on. Also, its almost like we are caches once without library and in platform implementation which seemed redundant.
However, need for retry seems like a valid question. But should we not do any further active queries at all? Or should we skip the exponential retries and have periodic active queries with larger intervals?

Reply:
I was wondering if Avahi doesn't do retries on its own. At least this is what I observe on wireshark when I run `avahi-browse` command. The retries must follow an exponential backoff per mDNS specification.

The problem with killing and restarting the browser is that there's a short period of time Avahi will not listen. Additionally, if it has some services cached, they will be provided to the application for each retry - is that right or am I missing something? Thus, if I understand correctly that Avahi does retries on its own, it would be easier and better to just keep it running.

Reply:
Yes, Avahi does have services cached which is provided for each retry. That is exactly why I wanted to clear the services cached in platform implementation here. 
I am not quite sure if avahi does retry on its own. Let me explore on this and get back.

Reply:
I commented all retry and re-attempt to querying. Looks like avahi takes care of exp. retries on its own and then some. The retries goes up exponentially till 64s and then queries every 97s +/- 1s.
As discussed offline,  removes all sorts of retries and letting avahi take care of retries.